File: src/main/java/top/dsbbs2/bukkitcord/utils/Text.java
package top.dsbbs2.bukkitcord.utils;

import java.util.*;

public class Text implements Comparable<Object> {
    private volatile String text;

    public Text() {
        this.text = "";
    }

    public Text(String t) {
        if(t == null) t = "";
        this.text = t.toLowerCase(Locale.ENGLISH);
    }

    public void setText(String t) {
        if(t == null) t = "";
        this.text = t.toLowerCase(Locale.ENGLISH);
    }

    public String getText() {
        if(text == null) text = "";
        return text.toLowerCase(Locale.ENGLISH);
    }
    @Override
    public String toString() {
        return getText();
    }
    @Override
    public boolean equals(Object o) {
        if(o instanceof Text) return ((Text)o).getText().equalsIgnoreCase(this.getText());
        if(o instanceof String) return ((String)o).equalsIgnoreCase(this.getText());
        if(o !=null ) return o.toString().equalsIgnoreCase(this.getText());
        return false;
    }
    @Override
    public int compareTo(Object o) {
        if(o instanceof Text) return ((Text)o).getText().toLowerCase().compareTo(this.getText().toLowerCase());
        if(o instanceof String) return ((String)o).toLowerCase().compareTo(this.getText().toLowerCase());
        if(o!=null) return this.getText().toLowerCase().compareTo(o.toString().toLowerCase());
        return 0;
    }
}

File: src/main/java/top/dsbbs2/bukkitcord/utils/Text.kt
package top.dsbbs2.bukkitcord.utils

import java.util.*

class Text : Comparable<Any?> {
    @Volatile
    private var text: String?

    constructor() {
        text = ""
    }

    constructor(t: String?) {
        var t = t
        if (t == null) t = ""
        text = t.toLowerCase(Locale.ENGLISH)
    }

    fun setText(t: String?) {
        var t = t
        if (t == null) t = ""
        text = t.toLowerCase(Locale.ENGLISH)
    }

    fun getText(): String {
        if (text == null) text = ""
        return text!!.toLowerCase(Locale.ENGLISH)
    }

    override fun toString(): String {
        return getText()
    }

    override fun equals(o: Any?): Boolean {
        return when (o) {
            is Text -> o.getText().equals(getText(), ignoreCase = true)
            is String -> o.equals(getText(), ignoreCase = true)
            else -> (o as? StringBuilder)?.toString()?.equals(getText(), ignoreCase = true)
                    ?: (o?.toString()?.equals(getText(), ignoreCase = true) ?: false)
        }
    }

    override fun compareTo(o: Any?): Int {
        return when (o) {
            is Text -> o.getText().compareTo(getText(), ignoreCase = true)
            is String -> o.compareTo(getText(), ignoreCase = true)
            else -> (o as? StringBuilder)?.toString()?.compareTo(getText(), ignoreCase = true)
                    ?: (o?.toString()?.compareTo(getText(), ignoreCase = true) ?: 0)
        }
    }
}